File: Data Structures & Algorithms/max-water-container/submission-6.java
class Solution {
    public int maxArea(int[] arr) {
        int left = 0;
        int right = arr.length - 1;
        int maxContainer = 0;
        while(left < right) {
            int areaContainer = Math.min(arr[left], arr[right]) * (right - left);
            maxContainer = Math.max(maxContainer, areaContainer);
            if(arr[left] < arr[right]) {
                left++;
            }
            else {
                right--;
            }
        }
        return maxContainer;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(arr, target) {
        let Left = 0;
        let Right = arr.length - 1;
        while(Left <= Right) {
            let Mid = Math.floor((Left + Right) / 2);
            if(arr[Mid] == target) {
                return Mid;
            }
            else if(arr[Left] <= arr[Mid]) {
                if(target < arr[Left] || target > arr[Mid]) {
                    Left = Mid + 1;
                }
                else {
                    Right = Mid - 1;
                }
            }
            else {
                if(target > arr[Right] || target < arr[Mid]) {
                    Right = Mid - 1;
                }
                else {
                    Left =  Mid + 1;
                }
            }
        }
        return - 1;
    }
}
